src/botmsg.c

```diff
@@ -812,14 +812,19 @@ int add_note(char *to, char *from, char *msg, int idx, int echo)
   }
 
   /* Might be form "sock:nick" */
-  splitc(ssf, from, ':');
+  splitcn(ssf, from, ':', sizeof ssf);
   rmspace(ssf);
-  splitc(ss, to, ':');
+  splitcn(ss, to, ':', sizeof ss);
   rmspace(ss);
   if (!ss[0])
     sock = -1;
-  else
-    sock = atoi(ss);
+  else {
+    sock = strtoul(ss, &endptr, 10);
```

Comment:
I would prefer a(n) (unsigned) long as temporary variable, do the range check, then cast to int.
And braces around this if/else construct

Reply:
Done.

Reply:
in case you wonder about any of those bits, i took it mostly from here: https://man.openbsd.org/strtol